File: questao_13/questao_13.go
package main

import (
	"fmt"
)

func Fatorial(n int) int {
	if n < 0 {
		panic("Fatorial não é definido para números negativos.")
	}
	if n == 0 || n == 1 {
		return 1
	}
	return n * Fatorial(n-1)
}

func main() {
	numero := 5
	resultado := Fatorial(numero)
	fmt.Printf("O fatorial de %d é %d.\n", numero, resultado)
}

File: questao_13/questao_13.py
# Métodos Estáticos: Adicione um método estático à classe Matematica que calcula o fatorial de um número. Em Python, utilize @staticmethod, em Java static, 
# e em Golang crie uma função regular na struct.

class Matematica:
    @staticmethod
    def fatorial(n):
        if n < 0:
            raise ValueError("Fatorial não é definido para números negativos.")
        if n == 0 or n == 1:
            return 1
        return n * Matematica.fatorial(n - 1)

numero = 5
resultado = Matematica.fatorial(numero)
print(f"O fatorial de {numero} é {resultado}.")
